cmd/oras/internal/option/remote.go

```diff
@@ -273,6 +281,11 @@ func (opts *Remote) NewRepository(reference string, common Common) (repo *remote
 	if repo.Client, err = opts.authClient(hostname, common.Debug); err != nil {
 		return nil, err
 	}
+	repo.HandleWarning = func(warning remote.Warning) {
+		if _, loaded := opts.hasWarned.LoadOrStore(warning.WarningValue, true); loaded {
```

Comment:
Should we warn only when the warning value is first stored? That it, `!loaded`?

Reply:
same applies to line 262.